src/main/fc/rc.c

```diff
@@ -496,134 +496,182 @@ NOINLINE void initAveraging(uint16_t feedforwardAveraging)
-    setpointAcceleration = setpointAcceleration * pid->feedforwardBoostFactor * jitterAttenuator * zeroTheAcceleration;
+    prevAcceleration[axis] = setpointAcceleration;
+
+    // apply gain factor to acceleration
+    setpointAcceleration *= rxRate * pid->feedforwardBoostFactor;
+
+    feedforward = setpointSpeed;
+
+    if (axis < FD_YAW) {
+        // for pitch and roll, add acceleration element (boost)
+        feedforward += setpointAcceleration;
+        // apply jitter reduction multiplier
+        feedforward *= jitterAttenuator;
+        // pull feedforward back towards zero as sticks approach max
+        if (pid->feedforwardMaxRateLimit) {
+            if (feedforward * setpoint > 0.0f) { // in same direction
+                const float limit = (maxRcRate[axis] - fabsf(setpoint)) * pid->feedforwardMaxRateLimit;
+                feedforward = (limit > 0.0f) ? constrainf(feedforward, -limit, limit) : 0.0f;
+            }
+        }
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 1, lrintf(setpointSpeed * 0.1f)); // base feedforward without acceleration
+    } else {
+        // for yaw, only apply jitter reduction only to the feedforward delta element
+        // can't be applied to the 'sustained' element since it will divide it to a small fraction of what is needed
+        feedforward *= jitterAttenuator;
+
+        // instead of adding setpoint acceleration, which is too aggressive,
+        // add a slow-fading high-pass filtered setpoint element
+        // this provides a 'sustained boost' with low noise
+        // it mimics the normal sustained yaw motor drive requirements, reducing P and I and hence reducing bounceback
+        // this doesn't add significant noise to feedforward
+        // too little yaw FF causes iTerm windup and slow bounce back when stopping a hard yaw
+        // too much causes fast bounce back when stopping a hard yaw
+
+        static float prevSetpointYaw;
+        // calculate lowpass gain factor from user specified time constant, can't exceed 1.0
+        const float kFY = fminf(rxInterval * pid->feedforwardYawHoldTime, 1.0f);
+        // subtract lowpass from input to get highpass
+        const float setpointLpfYaw = prevSetpointYaw + kFY * (setpoint - prevSetpointYaw);
+        prevSetpointYaw = setpointLpfYaw;
+        // provide separate boost gain for yaw; value of 20 works well for 5in, zero disables
+        const float feedforwardYawHold = pid->feedforwardYawHoldGain * (setpoint - setpointLpfYaw);
+
+        DEBUG_SET(DEBUG_FEEDFORWARD, 6, lrintf(feedforward * 0.01f));  // basic yaw feedforward without hold element
+        DEBUG_SET(DEBUG_FEEDFORWARD, 7, lrintf(feedforwardYawHold * 0.01f));  // yaw feedforward hold element
+
+        feedforward += feedforwardYawHold;
+        // NB: yaw doesn't need max rate limiting since it rarely overshoots
+    }
+
+    // apply feedforward transition, if configured. Archaic (better to use jitter reduction)
```

Comment:
(IMO this can be implemented by using PWL curve on feedforward (factoring other magic numbers into it)